packages/opentelemetry-baggage-span-processor/README.md

```diff
@@ -0,0 +1,47 @@
+# @uphold/opentelemetry-baggage-span-processor
+
+Package that sets all baggage entries as span attributes.
+
+## Installation
+
+```sh
+npm install @uphold/opentelemetry-baggage-span-processor
+```
+
+## Why?
+
+Setting all baggage entries as span attributes using OpenTelemetry ensures end-to-end traceability, simplifies analysis, and enhances observability in distributed systems. This approach provides consistent contextual information across spans, facilitating correlation, context propagation, and streamlined troubleshooting. It also enables flexibility in using analysis tools, improving operational efficiency and accelerating issue resolution.
+
+## Usage
+
+Add `BaggageSpanProcessor` as a `spanProcessor` to the `tracerProvider`. Here's how it looks if you are using the NodeSDK:
+
+```js
+import { BaggageSpanProcessor } from '@uphold/opentelemetry-baggage-span-processor';
+import { NodeTracerProvider } from '@opentelemetry/sdk-trace-node';
+import { ProxyTracerProvider, trace } from '@opentelemetry/api';
+
+const sdk = new NodeSDK({});
+
+// We can't rely on passing a custom `textMapPropagator` or `spanProcessor` when initializing the NodeSDK because it will be ignored unless `spanProcessor` or `traceExporter` are specified.
+// However, when those are specified, the `OTEL_EXPORTER_*` variables won't be respected.
+// To circumvent this issue, we wrap `start` and set those after.
+sdk.start = wrap(sdk.start, start => {
```

Comment:
Perhaps we could do it like:

```js
sdk.start();

const proxyTracerProvider = trace.getTracerProvider() as ProxyTracerProvider;
const tracerProvider = proxyTracerProvider.getDelegate() as NodeTracerProvider;

tracerProvider.addSpanProcessor(new BaggageSpanProcessor());
```

Reply:
But check if we can do this after calling `start()`.

Reply:
It doesn't seem to work.

https://github.com/open-telemetry/opentelemetry-js/blob/f2b447dad543ec57af17372411bf11b6e731f813/experimental/packages/opentelemetry-sdk-node/src/sdk.ts#L297